assignment/hyosungkim_proposal.md

```diff
@@ -0,0 +1,42 @@
+
+
+## Technologies used
+
+I will begin with the leaflet map, but ultimately plan to use Mapbox as the base for my map. I plan to use ajax calls for retrieving data, leaflet to include the functionality of adding and removing markers, turf to include the functionality of number of nearby rentals using a buffer, and underscore functions for iterating over rows of similar data.
+
+
+
+## Design spec
+
+#### User experience
+
+The main audience for my application is the Licenses and Inspection. Using this application will help them better prioritize where to conduct inspections given that there are a limited number of inspectors and a backlog of inspection units. Hopefully, this will also help them better identify where are the illegal rentals and in the bigger picture, help curb cases of illegal evictions.
+
+- Are there any website/application examples in the wild to which you can compare your final?
+
+#### Layouts and visual design
+
+The main component of the application will be a map of Philadelphia. I want to include a risk score slider so that users may be able to filter out high-risk, medium-risk and low-risk scored units. Because there are so many rental units, I want to include a functionality that aggregates the markers when the map is zoomed farther out. There will be a top bar with the title and a brief description of the application. I either plan to have a side bar or no side bar to display more details of each unit (number of bedrooms, license history, violation history, etc.) when clicked. With the additional buffer/count functionality, I want to include a checkbox to calculate the number of similar rentals in its vicinity when checked and not when not checked.
+
+
+## Anticipated difficulties
+
+The biggest challenge for this is the amount of data points to display on the map. My current data has over 300,000 points considered possible non-owner occupied rentals and I think that the processing for this will be very slow. A possible work around this issue is to narrow down on what I plan to display, for example, filtering out and displaying rentals with risk scores greater than x. I am still considering whether or not I will by default, filter the dataset or allow the user to input a range of risk scores to filter.
+
+
+## Missing pieces
+
+It would be useful to know if there was a way to continuously update a data file with inputs and although these wouldn't be used for the visual aspects of the application, it can serve to add data into the primary dataset and as next steps, use this information to improve the model. By updating the model every now and then, risk score predictions may improve.
```

Comment:
The only option for something like this would be a bit of back-end code that (nightly? hourly?) churns through inputs to update the model. The good news is that you can avoid that problem entirely for the sake of the final. If you need help pursuing next steps after that, let me know and I'll point you in the right direction